### PR TITLE
fix(VDataTable): add new headerProps  prop

### PR DESCRIPTION
**Pull request by websitevirtuoso**
extend ability to controll header props
https://github.com/vuetifyjs/vuetify/pull/18993

## Markup:

```vue

<template>
  <v-app>
    <v-container class="bg-pink">
      <v-row>
        <p>I expected the header to be pink like the table below</p>
        <v-data-table
          :items="cars"
          :height="200"
          sticky
          :header-props="{ class: 'bg-pink' }"
          class="bg-transparent"
        ></v-data-table>
      </v-row>
      <v-row>
        <p>OK</p>
        <v-data-table :items="cars" class="bg-transparent"></v-data-table>
      </v-row>
      <v-row>
        <div style="width: 100%">Virtual</div>
        <v-data-table-virtual
          :headers="headers"
          :items="virtualDesserts"
          class="elevation-1 bg-transparent"
          style="width: 400px"
          height="400"
          fixed-header
          item-value="name"
          :header-props="{ class: 'bg-pink' }"
        >
          <template #item.no="{index, item, internalItem, value}">
            {{internalItem.index + 1}}
          </template>
        </v-data-table-virtual>
      </v-row>
    </v-container>
  </v-app>
</template>

<script setup>
import { ref, computed } from 'vue'

const msg = ref('Hello World!')

const cars = [
  {
    name: 'Ford Mustang',
    horsepower: 450,
    fuel: 'Gasoline',
    origin: 'USA',
    price: 55000,
  },
  {
    name: 'Tesla Model S',
    horsepower: 670,
    fuel: 'Electric',
    origin: 'USA',
    price: 79999,
  },
  {
    name: 'BMW M3',
    horsepower: 503,
    fuel: 'Gasoline',
    origin: 'Germany',
    price: 70000,
  },
  {
    name: 'Audi RS6',
    horsepower: 591,
    fuel: 'Gasoline',
    origin: 'Germany',
    price: 109000,
  },
  {
    name: 'Chevrolet Camaro',
    horsepower: 650,
    fuel: 'Gasoline',
    origin: 'USA',
    price: 62000,
  },
  {
    name: 'Porsche 911',
    horsepower: 379,
    fuel: 'Gasoline',
    origin: 'Germany',
    price: 101000,
  },
  {
    name: 'Jaguar F-Type',
    horsepower: 575,
    fuel: 'Gasoline',
    origin: 'UK',
    price: 61000,
  },
  {
    name: 'Mazda MX-5',
    horsepower: 181,
    fuel: 'Gasoline',
    origin: 'Japan',
    price: 26000,
  },
  {
    name: 'Nissan GT-R',
    horsepower: 565,
    fuel: 'Gasoline',
    origin: 'Japan',
    price: 113540,
  },
  {
    name: 'Mercedes-AMG GT',
    horsepower: 523,
    fuel: 'Gasoline',
    origin: 'Germany',
    price: 115900,
  },
]

const headers = ref([
  {
    title: 'No',
    align: 'start',
    sortable: false,
    key: 'no',
    fixed: true,
    headerProps: { class: 'bg-blue' },
    cellProps: () => ({
      class: 'text-warning bg-blue',
    }),
  },
  {
    title: 'Dessert (100g serving)',
    align: 'start',
    sortable: false,
    key: 'name',
    fixed: true,
    cellProps: () => ({
      class: 'text-black',
    }),
  },
  {
    title: 'Calories',
    align: 'end',
    key: 'calories',
    headerProps: { class: 'bg-white text-black' },
  },
  { title: 'Fat (g)', align: 'end', key: 'fat' },
  { title: 'Carbs (g)', align: 'end', key: 'carbs' },
  { title: 'Protein (g)', align: 'end', key: 'protein' },
  {
    title: 'Iron (%)',
    align: 'end',
    key: 'iron',
    fixedOffset: 100,
    lastFixed: true,
  },
])
const desserts = ref([
  {
    name: 'Frozen Yogurt',
    calories: 159,
    fat: 6.0,
    carbs: 24,
    protein: 4.0,
    iron: '1',
  },
  {
    name: 'Ice cream sandwich',
    calories: 237,
    fat: 9.0,
    carbs: 37,
    protein: 4.3,
    iron: '1',
  },
  {
    name: 'Eclair',
    calories: 262,
    fat: 16.0,
    carbs: 23,
    protein: 6.0,
    iron: '7',
  },
  {
    name: 'Cupcake',
    calories: 305,
    fat: 3.7,
    carbs: 67,
    protein: 4.3,
    iron: '8',
  },
  {
    name: 'Gingerbread',
    calories: 356,
    fat: 16.0,
    carbs: 49,
    protein: 3.9,
    iron: '16',
  },
  {
    name: 'Jelly bean',
    calories: 375,
    fat: 0.0,
    carbs: 94,
    protein: 0.0,
    iron: '0',
  },
  {
    name: 'Lollipop',
    calories: 392,
    fat: 0.2,
    carbs: 98,
    protein: 0,
    iron: '2',
  },
  {
    name: 'Honeycomb',
    calories: 408,
    fat: 3.2,
    carbs: 87,
    protein: 6.5,
    iron: '45',
  },
  {
    name: 'Donut',
    calories: 452,
    fat: 25.0,
    carbs: 51,
    protein: 4.9,
    iron: '22',
  },
  {
    name: 'KitKat',
    calories: 518,
    fat: 26.0,
    carbs: 65,
    protein: 7,
    iron: '6',
  },
])

const virtualDesserts = computed(() => {
  return [...Array(100).keys()].map(i => {
    const dessert = { ...desserts.value[i % desserts.value.length] }
    dessert.name = `${dessert.name} #${i + 1}`
    return dessert
  })
})
</script>
```

### CODE REVIEW

File: packages/vuetify/src/components/VDataTable/VDataTableHeaders.tsx [hunk 1]
```diff
@@ -138,7 +141,10 @@ export const VDataTableHeaders = genericComponent<VDataTableHeadersSlots>()({
           fixed={ column.fixed }
           lastFixed={ column.lastFixed }
           noPadding={ noPadding }
-          { ...column.headerProps }
+          {
+            // column.headerProps have more priorities over props
+            ...Object.assign({}, props.headerProps || {}, column.headerProps || {})
```
Comment (johnleider) [inline packages/vuetify/src/components/VDataTable/VDataTableHeaders.tsx:146]: use mergeDeep instead

**Review by johnleider — CHANGES_REQUESTED** [review]

File: packages/vuetify/src/components/VDataTable/VDataTableHeaders.tsx (hunk 1 above)
Comment (websitevirtuoso) [reply packages/vuetify/src/components/VDataTable/VDataTableHeaders.tsx]: Fixed. Thanks

Comment (KaelWD) [reply packages/vuetify/src/components/VDataTable/VDataTableHeaders.tsx]: mergeProps not mergeDeep

**Review by johnleider — APPROVED** [review]

**Comment by nehalshah50** [comment]
When is this fix available. I am setting custom class property using headerProps but its not taking it.

**Comment by KaelWD** [comment]
image

**Comment by nehalshah50** [comment]
> image
> 

I am on the latest release and it's not working

**Pull request merged**